File: Java_MVVM/app/src/main/java/eu/gotoinc/requesinjava_mvvm/repository/ApiContract.java
package eu.gotoinc.requesinjava_mvvm.repository;


import java.util.List;

import eu.gotoinc.requesinjava_mvvm.repository.data_models.responses.PaginationResponse;
import eu.gotoinc.requesinjava_mvvm.repository.data_models.responses.UserResponse;
import io.reactivex.Single;

public interface ApiContract {
    Single<PaginationResponse<List<UserResponse>>> getUsersList(int page);
}

File: Kotlin_MVVM/app/src/main/java/eu/gotoinc/kotlinmvvm/repository/ApiContract.kt
package eu.gotoinc.kotlinmvvm.repository

import eu.gotoinc.kotlinmvvm.repository.datamodel.response.PaginationResponse
import eu.gotoinc.kotlinmvvm.repository.datamodel.response.UserResponse
import io.reactivex.Single

interface ApiContract {
    fun getUsersList(page: Int): Single<PaginationResponse<List<UserResponse>>>
}